frontend/src/utils/stringUtils.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Many Translator services now have differently formatted types. These utilities
+ * should help keep everything consistent internally.
+ * Incoming biolink types are space case
+ * All outgoing types are 'biolink:PascalCase'
+ * User input types can be anything
+ */
+import _ from 'lodash';
+
+function toSpaceCase(str) {
+  return _.startCase(str);
+}
+
+function toCamelCase(str) {
+  return _.camelCase(str);
+}
+
+function toSnakeCase(str) {
+  return _.snakeCase(str);
+}
+
+/**
+ * Convert to pascal case with biolink curie format
+ * @param {string} str string to convert to pascal case
+ */
+function toPascalCase(str) {
+  const camelCaseStr = _.camelCase(str);
+  const pascalType = `${camelCaseStr.charAt(0).toUpperCase()}${camelCaseStr.slice(1)}`;
```

Comment:
Pascal case conversion doesn't work for terms like RNAProduct. To create a node with RNAProduct, you have to type "Rna Product" and then if you download it the query graph has "biolink:RnaProduct" which isn't correct. I think we may need a custom exception list.

Reply:
I have discussed this with Patrick. Currently, nothing comes with an RNA type, so for now I think it's fine to keep it as is. Later on if nodes get added with that type, we'll be able to see how they handle it and we can update our conversions.